crates/threshold-bls/src/sig/sig.rs

```diff
@@ -85,25 +89,38 @@ pub type Partial = Vec<u8>;
 /// signature" to then produce a regular signature.
 /// The `dkg` module allows participants to create a distributed private/public key
 /// that can be used with implementations `ThresholdScheme`.
-pub trait ThresholdScheme: Scheme {
-    fn partial_sign(private: &Share<Self::Private>, msg: &[u8]) -> Result<Partial, Box<dyn Error>>;
+pub trait ThresholdScheme: SignatureScheme {
```

Comment:
The reason I didn't go with that approach in the first place is that ```thesholdScheme.Sign()``` is **not** a valid function a user should use: it has no meaning by itself. I searched if it was possible to disable one func but it's not hence i went this way.
So I wouldn't go this way, even though it's simpler / more elegant for the code. User API is more important imo.

Reply:
Or maybe you make it a second Threshold trait that is exposed doesn't contain the `Sign()` method and implement this trait automatically for all `ThresholdScheme`. But it sounds like not super useful use of time tbh.

Reply:
This is a good point, thank you. I have added https://github.com/celo-org/celo-threshold-bls-rs/pull/30/commits/0e3cf0663509bda7649f103afa44f63f4894851e, which should address this.